### PR TITLE
Informer and refactor

### PR DESCRIPTION
**Pull request by kimikowang**

### CODE REVIEW

**Comment by kimikowang** [comment]
@kalantar @fabolive The cluster used in travis-ci is broken. So the tests can not be run. I ran them locally with my own cluster and all tests passed.

File: pkg/analytics/metrics/configmap.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	iter8v1alpha1 "github.com/iter8-tools/iter8-controller/pkg/apis/iter8/v1alpha1"
+)
+
+const (
+	MetricsConfigMap = "iter8config-metrics"
+	Iter8Namespace   = "iter8"
```
Comment (kalantar) [inline pkg/analytics/metrics/configmap.go:31]: We are hard-coding the namespace. For RedHat scenarios we will need to make this configurable.

Comment (kimikowang) [reply pkg/analytics/metrics/configmap.go:31]: Yes, these are the default values only. A more flexible setting will be added later.

File: pkg/controller/experiment/experiment_controller.go [hunk 2]
```diff
@@ -226,49 +313,91 @@ func (r *ReconcileExperiment) Reconcile(request reconcile.Request) (reconcile.Re
...
+	if err := r.syncMetrics(ctx, instance); err != nil {
+		return reconcile.Result{}, nil
 	}
 
+	apiVersion := instance.Spec.TargetService.APIVersion
+
+	switch apiVersion {
+	case KubernetesService:
+		return r.syncKubernetes(ctx, instance)
+	case KnativeServiceV1Alpha1:
```
[the remark this answers is not in the document]
Comment (kimikowang) [reply pkg/controller/experiment/experiment_controller.go:339]: I am removing this in the later pr.

File: pkg/apis/iter8/v1alpha1/experiment_types.go [hunk 3]
```diff
@@ -149,8 +105,8 @@ type TargetService struct {
 type Phase string
 
 const (
-	// PhaseInitializing indicates experiment is intializing
-	PhaseInitializing Phase = "Initializing"
+	// PhaseInit indicates experiment is being initializing
+	PhaseInit Phase = "Init"
```
[the remark this answers is not in the document]
Comment (kimikowang) [reply pkg/apis/iter8/v1alpha1/experiment_types.go]: Updated. removed.

File: install/iter8-controller.yaml [hunk 4]
```diff
@@ -155,6 +155,13 @@ spec:
         spec:
           description: ExperimentSpec defines the desired state of Experiment
           properties:
+            action:
+              description: Action specifies the action that the user would like to
+                perform
+              enum:
+              - resume
+              - pause
+              type: string
```
[the remark this answers is not in the document]
Comment (kimikowang) [reply install/iter8-controller.yaml]: Updated.

File: pkg/controller/experiment/experiment.go [hunk 5]
```diff
@@ -463,3 +470,28 @@ func (r *ReconcileExperiment) startNextIteration(context context.Context, instan
 	r.MarkExperimentProgress(context, instance, false, iter8v1alpha1.ReasonIterationUpdate,
 		"Iteration %d Started", instance.Status.CurrentIteration)
 }
+
+func updateSubset(dr *v1alpha3.DestinationRule, d *appsv1.Deployment, name string) bool {
```
[the remark this answers is not in the document]
Comment (kimikowang) [reply pkg/controller/experiment/experiment.go:474]: The routing part is not fully-refactored yet, so some rules-manipulation functions are still left in the `experiment.go`.  These can be fixed in future routing-refactor work.

File: pkg/apis/iter8/v1alpha1/experiment_types.go [hunk 6]
```diff
@@ -45,6 +45,7 @@ type Experiment struct {
 	Spec    ExperimentSpec    `json:"spec,omitempty"`
 	Status  ExperimentStatus  `json:"status,omitempty"`
 	Metrics ExperimentMetrics `json:"metrics,omitempty"`
+	Action  ExperimentAction  `json:"action,omitempty"`
```
[the remark this answers is not in the document]
Comment (kimikowang) [reply pkg/apis/iter8/v1alpha1/experiment_types.go]: Nice catch. Pushed the fix.

**Pull request merged**